Include/cpython/object.h

```diff
@@ -538,7 +538,7 @@ PyAPI_FUNC(void) _PyTrash_end(struct _ts *tstate);
 
 #define Py_TRASHCAN_BEGIN(op, dealloc) \
     Py_TRASHCAN_BEGIN_CONDITION(op, \
-        Py_TYPE(op)->tp_dealloc == (destructor)(dealloc))
+        PyType_GetSlot(Py_TYPE(op), Py_tp_dealloc) == (destructor)(dealloc))
```

Comment:
If we introduce a function call in this macro, I would prefer to add a _PyTrash_cond(op, dealloc) function. This function could access directly the PyTypeObject.tp_dealloc member. Something similar to _PyTrash_begin() and _PyTrash_end() private functions.

Reply:
Do you means doing something like:
```
PyAPI_FUNC(void) _PyTrash_cond(PyObject *op, destructor dealloc);

#define Py_TRASHCAN_BEGIN(op, dealloc) \
    _PyTrash_cond(_PyObject_CAST(op), (destructor)dealloc);
```

Reply:
Looks like we can not call the `_PyTrash_cond` function in `Py_TRASHCAN_BEGIN`.
Because `Py_TRASHCAN_BEGIN` macro company with `Py_TRASHCAN_END` mcaro all the time which defined in `Py_TRASHCAN_BEGIN_CONDITION` macro. something like: https://github.com/python/cpython/blob/master/Objects/tupleobject.c#L272-L300

Reply:
No. I mean something like:

```
#define Py_TRASHCAN_BEGIN(op, dealloc) \
    Py_TRASHCAN_BEGIN_CONDITION(op, _Py_trashcan_cond(op, dealloc))
```

Only modify the second argument of the Py_TRASHCAN_BEGIN_CONDITION() macro.

Reply:
oh. Got it.